pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +354,132 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Namespace is the source namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the source pod.
+	Pod string `json:"pod"`
+	// IP is the source IPv4 or IPv6 address.
+	IP string `json:"ip"`
+}
+
+// Destination describes the destination spec of the packetcapture.
+type Destination struct {
+	// Namespace is the destination namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the destination pod, exclusive with destination service.
+	Pod string `json:"pod"`
+	// Service is the destination service, exclusive with destination pod.
+	Service string `json:"service"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP string `json:"ip"`
+}
+
+// IPv6Header describes spec of an IPv6 header.
+type IPv6Header struct {
+	// NextHeader is the IPv6 protocol.
+	NextHeader *int32 `json:"nextHeader,omitempty" yaml:"nextHeader,omitempty"`
+	// HopLimit is the IPv6 Hop Limit.
+	HopLimit *int32 `json:"hopLimit,omitempty" yaml:"hopLimit,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMP *ICMPEchoRequestHeader `json:"icmp,omitempty" yaml:"icmp,omitempty"`
+	UDP  *UDPHeader             `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP  *TCPHeader             `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	IPv6Header      *IPv6Header     `json:"ipv6Header,omitempty"`
+	TransportHeader TransportHeader `json:"transportHeader"`
+}
+
+// PacketCaptureFirstNConfig contains the config for the FirstN type capture. The only supported parameter is
+// `Number` at the moment, meaning capturing the first specified number of packets in a flow.
+type PacketCaptureFirstNConfig struct {
+	Number int32 `json:"number"`
+}
+
+const DefaultPacketCaptureTimeout uint16 = 60
+
+type PacketCapturePhase string
+
+const (
```

Comment:
also see discussion https://github.com/antrea-io/antrea/pull/6257#discussion_r1600696782

for the constants part i think it's unnecessary,  each CRD may have more different status in the future.

About types, since `Traceflow` is in `vebeta1` now, are you suggesting we define some common types in neutral place?

Reply:
I am not sure if we should define common types across versions. Let us see if @tnqn and @antoninbas have an opinion.

Reply:
I guess in theory we could have common definitions in a neutral directory, it wouldn't need to be under `apis/crd/<version>`. But if it's not a versioned directory, then we would have to be confident that they are not going to change.

K8s has `meta/v1` with comm types, such as `Time`. I don't know how good of an analogy it is.

Reply:
Except those meta types in meta/v1. K8s doesn't define common types across versions. If a constant needs to be used by a version, it will just be replicated in that version. I think it's to keep every version's stability and compatability. Otherwise a change in one version would have an impact on other versions.